### PR TITLE
use deployment event ID as unique identifier

### PR DESCRIPTION
**Pull request by xh3b4sd**
While hustling through my oncall shift I noticed it is not possible with the eventer/operator to redeploy the very same ref. This PR fixes the eventer side.

### CODE REVIEW

File: service/informer/informer.go [hunk 1]
```diff
@@ -227,6 +190,33 @@ func (s *Service) bootWithError() error {
...
+		return microerror.Mask(err)
+	}
+
+	err = s.eventer.SetPendingStatus(e)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
```
Comment (xh3b4sd) [inline service/informer/informer.go:218]: This is refactoring to make the code more DRY.

File: service/informer/informer.go [hunk 2]
```diff
@@ -253,7 +243,7 @@ func ensureProject(projects []draughtsmantprspec.Project, project draughtsmantpr
 	}
 
 	for i, p := range projects {
-		if p.Name == project.Name && p.Ref != project.Ref {
+		if p.Name == project.Name && p.ID != project.ID {
```
Comment (xh3b4sd) [inline service/informer/informer.go:246]: This is the actual fix.

File: service/informer/informer_test.go [hunk 3]
```diff
@@ -434,15 +434,16 @@ func Test_Informer_ensureProject(t *testing.T) {
 		ExpectedProjects []draughtsmantprspec.Project
 		ExpectedUpdated  bool
 	}{
-		// Test 1
+		// Test 1 ensures that zero value input results in zero value output.
```
Comment (xh3b4sd) [inline service/informer/informer_test.go:437]: I improved the comments here. I was lazy back then.

File: service/informer/informer_test.go [hunk 4]
```diff
@@ -523,6 +525,31 @@ func Test_Informer_ensureProject(t *testing.T) {
...
+			},
+			ExpectedProjects: []draughtsmantprspec.Project{
+				{
+					ID:   "api-id-2",
+					Name: "api-name",
+					Ref:  "api-sha-1",
+				},
+			},
+			ExpectedUpdated: true,
+		},
```
Comment (xh3b4sd) [inline service/informer/informer_test.go:552]: This new test shows what this changeset does.

File: service/informer/informer_test.go [hunk 5]
```diff
@@ -490,7 +491,8 @@ func Test_Informer_ensureProject(t *testing.T) {
 			ExpectedUpdated: true,
 		},
 
-		// Test 4
+		// Test 4 that a different ref causes a deployment update of the right
+		// project.
```
Comment (rossf7) [inline service/informer/informer_test.go]: Nitpick but this comment missed ensures.

**Review by rossf7 — APPROVED** [review]
Not very familiar with the eventer but change LGTM.

File: service/informer/informer_test.go (hunk 5 above)
Comment (xh3b4sd) [reply service/informer/informer_test.go]: Fixed.

**Pull request merged**